### PR TITLE
docs: git auth

### LINKED ISSUE #4771: docs: git providers
You can configure Git OAuth via environment variables:

```sh
CODER_GITAUTH_0_TYPE=github|gitlab|azure-devops|bitbucket
CODER_GITAUTH_0_CLIENT_ID=xxxxxx
CODER_GITAUTH_0_CLIENT_SECRET=xxxxxxx

For your OAuth application, set the callback URL to:
/gitauth/<type>/callback
```

### PR DESCRIPTION
**Pull request by bpmct**
closes #4771

### CODE REVIEW

File: docs/admin/git-providers.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+## Configuration
+
+To add a git provider, you'll need to create an OAuth application. The following providers are supported:
+
+- [GitHub](https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app) (GitHub apps are also supported)
+- [GitLab](https://docs.gitlab.com/ee/integration/oauth_provider.html)
+- [BitBucket](https://support.atlassian.com/bitbucket-cloud/docs/use-oauth-on-bitbucket-cloud/)
+- [Azure DevOps](https://learn.microsoft.com/en-us/azure/devops/integrate/get-started/authentication/oauth?view=azure-devops)
+
+Example callback URL: `https://coder.example.com/api/v2/gitauth/primary-github/callback`. Use an arbitrary ID for your provider (e.g. `primary-github`).
```
Comment (bpmct) [inline docs/admin/git-providers.md:23]: @kylecarbs what URL do you use if an ID is not set? I was a bit confused

Comment (kylecarbs) [reply docs/admin/git-providers.md:23]: It'll be `/gitauth/<id but falls back to type>/callback`

**Review by ammario — APPROVED** [review]

File: docs/admin/git-providers.md (hunk 1 above)
Comment (kylecarbs) [reply docs/admin/git-providers.md:23]: btw @bpmct it's not `/api/v2`. I moved the callback to the root for a nicer route.

Comment (ghuntley) [inline docs/admin/git-providers.md:23]:
```suggestion
Example callback URL: `https://coder.example.com/gitauth/primary-github/callback`. Use an arbitrary ID for your provider (e.g. `primary-github`).
```

**Pull request merged**